core/src/test/java/com/exadel/aem/toolkit/core/injectors/DelegateInjector.java

```diff
@@ -62,39 +68,89 @@ public Object getValue(
         String effectiveName = StringUtils.defaultIfBlank(requestProperty.name(), name);
 
         if (delegate instanceof RequestAttributeInjector) {
-            return ((RequestAttributeInjector) delegate).getValue(
+            Object value = ((RequestAttributeInjector) delegate).getValue(
                 AdaptationUtil.getRequest(adaptable),
                 effectiveName,
                 type);
+            boolean isFallbackValue = value instanceof CastResult && ((CastResult) value).isFallback();
```

Comment:
❗ I didn't quite get  why you are duplicating the logic that must have been in the `BaseInjector` to this tests case. This is so wrong. We must only test it here, not paste it here. It needs to be tested. But when it is here it looks like this logic is true by default